alvr/filesystem/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+}
+
+// Layout of the ALVR installation. All paths are absolute
+pub struct Layout {
+    // directory containing the launcher executable
+    pub user_executables_dir: PathBuf,
+    // parent directory of resources like the dashboard and presets folders
+    pub static_resources_dir: PathBuf,
+    // directory for storing configuration files (session.json)
+    pub config_dir: PathBuf,
+    // directory for storing log
+    pub log_dir: PathBuf,
+    // directory to register in openVR driver path
+    pub openvr_driver_dir: PathBuf,
+    // (linux only) parent directory of the executable to wrap vrcompositor
+    pub vrcompositor_wrapper_dir: PathBuf,
+}
+
+impl Layout {
+    // This constructor should be used directly only for development builds and packaging
+    pub fn from_common_ancestor(path: &Path) -> Self {
+        if cfg!(any(windows, target_os = "macos")) {
+            Self {
+                user_executables_dir: path.to_owned(),
+                static_resources_dir: path.to_owned(),
+                config_dir: path.to_owned(),
+                log_dir: path.to_owned(),
+                openvr_driver_dir: path.to_owned(),
+                vrcompositor_wrapper_dir: PathBuf::new(),
+            }
+        } else if cfg!(target_os = "linux") {
+            Self {
+                user_executables_dir: path.join("bin"),
+                static_resources_dir: path.join("share/alvr"),
+                config_dir: dirs::config_dir().unwrap().join("alvr"),
+                log_dir: dirs::home_dir().unwrap(),
+                openvr_driver_dir: path.join("lib64/alvr"),
+                vrcompositor_wrapper_dir: path.join("libexec/alvr"),
+            }
+        } else {
+            unimplemented!()
+        }
+    }
+
+    // The path should include the executable file name
+    pub fn from_launcher_exe(path: &Path) -> Self {
+        let ancestor = if cfg!(any(windows, target_os = "macos")) {
+            path.parent().unwrap()
+        } else if cfg!(target_os = "linux") {
+            path.parent().unwrap().parent().unwrap()
+        } else {
+            unimplemented!()
+        };
+
+        Self::from_common_ancestor(ancestor)
+    }
+
+    pub fn from_openvr_driver_dir(dir: &Path) -> Self {
+        let ancestor = if cfg!(windows) || cfg!(target_os = "macos") {
+            dir
+        } else if cfg!(target_os = "linux") {
+            dir.parent().unwrap().parent().unwrap()
+        } else {
+            unimplemented!()
+        };
+
+        Self::from_common_ancestor(ancestor)
+    }
+
+    pub fn from_default_installation() -> Self {
+        let ancestor = if cfg!(windows) {
+            PathBuf::from(r"C:\Program Files\ALVR")
+        } else if cfg!(target_os = "linux") {
+            PathBuf::from("/")
+        } else if cfg!(target_os = "macos") {
+            env::temp_dir()
+        } else {
+            unimplemented!()
+        };
+
+        Self::from_common_ancestor(&ancestor)
+    }
```

Comment:
This function is unused, and if it were, we need to support at least `/usr` and `/usr/local`

Reply:
It was just an idea, if there is no use i'll delete it.